packages/contracts-bedrock/src/L1/SystemConfig.sol

```diff
@@ -149,6 +209,62 @@ contract SystemConfig is OwnableUpgradeable, ISemver {
         addr_ = Storage.getAddress(UNSAFE_BLOCK_SIGNER_SLOT);
     }
 
+    /// @notice Getter for the L1CrossDomainMessenger address.
+    function l1CrossDomainMessenger() external view returns (address addr_) {
+        addr_ = Storage.getAddress(L1_CROSS_DOMAIN_MESSENGER_SLOT);
+    }
+
+    /// @notice Getter for the L1ERC721Bridge address.
+    function l1ERC721Bridge() external view returns (address addr_) {
+        addr_ = Storage.getAddress(L1_ERC_721_BRIDGE_SLOT);
+    }
+
+    /// @notice Getter for the L1StandardBridge address.
+    function l1StandardBridge() external view returns (address addr_) {
+        addr_ = Storage.getAddress(L1_STANDARD_BRIDGE_SLOT);
+    }
+
+    /// @notice Getter for the L2OutputOracle address.
+    function l2OutputOracle() external view returns (address addr_) {
+        addr_ = Storage.getAddress(L2_OUTPUT_ORACLE_SLOT);
+    }
+
+    /// @notice Getter for the OptimismPortal address.
+    function optimismPortal() external view returns (address addr_) {
+        addr_ = Storage.getAddress(OPTIMISM_PORTAL_SLOT);
+    }
+
+    /// @notice Getter for the OptimismMintableERC20Factory address.
+    function optimismMintableERC20Factory() external view returns (address addr_) {
+        addr_ = Storage.getAddress(OPTIMISM_MINTABLE_ERC20_FACTORY_SLOT);
+    }
+
+    /// @notice Getter for the BatchInbox address.
+    function batchInbox() external view returns (address addr_) {
+        addr_ = Storage.getAddress(BATCH_INBOX_SLOT);
+    }
+
+    /// @notice Sets the start block in a backwards compatible way. Proxies
+    ///         that were initialized before the startBlock existed in storage
+    ///         can have their start block set by a user provided override.
+    ///         A start block of 0 indicates that there is no override and the
+    ///         start block will be set by `block.number`.
+    /// @dev    This logic is used to patch legacy with new storage values. In the
+    ///         next version, it should remove the override and set the start block
+    ///         to `block.number` if the value in storage is 0. This will allow it
+    ///         to be reinitialized again and also work for fresh deployments.
+    /// @param  _startBlock The start block override to set in storage.
+    function _setStartBlock(uint256 _startBlock) internal {
+        require(startBlock == 0, "SystemConfig: cannot override an already set start block");
```

Comment:
The following commit updated the logic here: https://github.com/ethereum-optimism/optimism/commit/13719bab4e1273cbdc418a278a14409768b13a6a

The `require` statement is broken because it prevents the contract from being initialized twice. Could we simplify this even more than what is in the above commit such that it does the following?

```solidity
if (startBlock == 0) {
  startBlock = block.number
}
```

Then as part of the upgrade, we generate a 2 step using `StorageSetter` that sets the value for legacy chains. It will be `type(uint256).max` in the constructor, so it will be a dead value in the singleton. It removes the risk of accidentally overriding the value again when the contract is reinitialized. We can then remove the argument to the function

Reply:
This still needs to be addressed

Reply:
addressed the changes to `SystemConfig.sol`. should we make any changes on the upgrade logic?

Reply:
We shouldn't need any changes to the upgrade logic